src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package pl.byczazagroda.trackexpensesappbackend.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryUpdateDTO;
+import pl.byczazagroda.trackexpensesappbackend.exception.AppRuntimeException;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+import pl.byczazagroda.trackexpensesappbackend.mapper.FinancialTransactionCategoryModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionCategoryRepository;
+
+import javax.transaction.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class FinancialTransactionCategoryServiceImpl implements FinancialTransactionCategoryService {
+
+    private final FinancialTransactionCategoryRepository financialTransactionCategoryRepository;
+
+    private final FinancialTransactionCategoryModelMapper financialTransactionCategoryModelMapper;
+
+    @Override
+    @Transactional
+    public FinancialTransactionCategoryDTO updateFinancialTransactionCategory(
+            Long id, FinancialTransactionCategoryUpdateDTO financialTransactionCategoryUpdateDTO) {
+        FinancialTransactionCategory financialTransactionCategory
+                = financialTransactionCategoryRepository.findById(id)
+                .orElseThrow(() -> new AppRuntimeException(ErrorCode.FTC001,
+                        String.format("Financial transaction category with id: %d not found", id)));
+        financialTransactionCategory.setName(financialTransactionCategoryUpdateDTO.name());
+        financialTransactionCategory.setType(financialTransactionCategoryUpdateDTO.type());
+        FinancialTransactionCategory updated
```

Comment:
What mean "update"? - maybe a better name for exactly this variable will be:
- entity,
- updatedEntity